### ISSUE TITLE
EPub is rendered different in Readium 2 than with Readium 1

### ISSUE THREAD

**Issue by chrfalch** [readium/kotlin-toolkit#21]
## Bug Report

We have an epub that was rendered fine with Readium 1, but after upgrading to Readium 2 it renders really strange.

### What happened?

Rendered in Readium 1:
![image](https://user-images.githubusercontent.com/875252/137150630-739a2fce-7fbc-4af2-b8ce-dcde1b96f0d3.png)

Rendered i Readium 2:
![image](https://user-images.githubusercontent.com/875252/137150689-75f00817-f31d-4f72-9d87-4402ff7760f5.png)

The publication has been shared through private channel

**Comment by chrfalch** [comment]
Thanks so much for the feedback again! We're still only on 2.0.0 and not over in Monorepo-land (getting very close to release so we're a bit limited), but I'll try to add this fixes and verify if it solves the issue during the weekend. And then we'll schedule an upgrade after the next release.

**Comment by mickael-menu** [comment]
Sounds good, let me know how it goes porting the fix.

**Comment by chrfalch** [comment]
Will it be enough to add the changes to the files `readium/navigator/src/main/assets/_scripts/src/index-reflowable.js` and `readium/streamer/src/main/java/org/readium/r2/streamer/fetcher/HtmlInjector.kt` to verify?

**Comment by chrfalch** [comment]
I'm not able to test this. Everything works fine after upgrading to the monorepo, but when I'm trying to use a local directory I'm getting into all kind of troubles with Android Studio. I have changed all references so that my app depends on the local files.

When building I get an error about a missing method (lowercase) in the XmlPullParser library. After upgrading Kotlin from 1.4.31 -> 1.5.31 this error goes away.

The next error I'm getting is a problem with a `kapt` build task for readium lcp that fails without any error message. Maybe there are more dependencies that I'm not aware of? Have tried adding the kapt plugin: through `apply plugin: "kotlin-kapt"` in my gradle file without any luck. Have also added a dependency on the room database - with no luck :(

**Comment by mickael-menu** [comment]
Did you upgrade to Readium 2.1.0 from the multiple repos first? I want to rule out any issue with the monorepo, as it should be the same code. For 2.1.0 you need to update Kotlin but also to Gradle 7.

**Comment by chrfalch** [comment]
Yes, it was working with the monorepo directly without upgrading anything else than the dependencies (as described in the readme)

**Comment by chrfalch** [comment]
Seems to be compiling correctly, changed the readium-version directly to

```readium_version = 'fix~swipes-SNAPSHOT'```

Would that be sufficient?

**Comment by mickael-menu** [comment]
Yes I think that should work.

**Comment by chrfalch** [comment]
Tried building with the setup above but it doesn't seem like the changes in your fix is deployed. Inspected the running app's webviews in chrome and the file `index-reflowable.js` does not contain the function you added.

**Comment by stevenzeck** [comment]
@chrfalch I think you want the `fix~meta-override-SNAPSHOT` one.

**Comment by chrfalch** [comment]
It is now working perfect. Thanks a million to @stevenzeck and @mickael-menu for helping out here!!!!

Any ETA on when a fix like this will be released? 

<img width="304" alt="image" src="https://user-images.githubusercontent.com/875252/137681139-f8063298-91d9-479c-89ed-07611835620a.png">
